cpp/README.md

```diff
@@ -0,0 +1,62 @@
+# Lance: A Columnar File Format
+
+
+# Development
+
+## Code Style
+
+The core of `Lance` data format is developed in `C++20`.
+
+The code base follows Apache Arrow and [Google C++ Code Style](https://google.github.io/styleguide/cppguide.html).
+With exceptions:
+* Line width: 100
+
+The code style is enforced via [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html). Please make sure `clang-format` has been set up appropriately in the IDE.
+
+## Build
+
+The code is developed on `macOS 12+` and a recent Linux (`Ubuntu 22.04`). Please file bugs or contribute fixes if you encounter issues with other Linux flavors.
+
+A recent C++ compiler that supports C++20 is needed. The toolchain that has been tested is:
+
+* AppleClang >= 13.1.6
+* GCC >= 11.2
+* CMake >= 3.22
+* Apache Arrow >= 8.0
+* Protobuf > 3.12
+
+```sh
+# On macOS 12+
+brew install apache-arrow cmake protobuf
+# Optionally, build document
+brew install doxygen
+```
+
+```sh
+# On Ubuntu 22.04
+# Follow https://arrow.apache.org/install/
+sudo apt update
+sudo apt install -y -V ca-certificates lsb-release wget
+wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+sudo apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+sudo apt update
+
+# Install arrow, protobuf, cmake
+sudo apt install -y -V libarrow-dev libarrow-dataset-dev libparquet-dev libarrow-python-dev cmake libprotobuf-dev protobuf-compiler
```

Comment:
Do we need to install protobuf-compiler explicitly? I forgot if it's a libprotobuf-dev dep

Reply:
Good call. lemme check